Fix networking issue on vm creation with vmware (closed bad merge)

almoore opened the pull request:
### What does this PR do?
Reference the container while creating device spec for vmware in salt cloud.
Implement a method to postgres pgjsonb for filtering.
### What issues does this PR fix or reference?
If two dvs devices (Distributed Virtual Switch) under the same virtual host for a vmware have the same name for a Distributed Virtual Port group then the wrong network adapter can be selected in the request to create a vm.  This will cause the vm to come up without a proper network switch device selected, effectively having no network. 

### Tests written?
No

Commit:
Activate jid_queue also for SingleMinions to workaround 0mq reconnection issues

Commit:
Add missing delete_on_termination passthrough. Adapt docs.

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
Don't allow path separators in minion ID

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

terminalmage committed:
Do not allow IDs with null bytes in decoded payloads

twangboy committed:
Add support for multimaster setup

Allows you to pass a comma-delimited list of masters to the master combo
box or the /master command line switch
Parses the existing minion config file to get multimaster settings
Adds an checkbox to use the existing config in the minion config page
When the box is checked, show the existing config grayed out
When unchecked, show default values
Adss the /use-existing-config= command line switch for use from the
command line
Adds support for a help switch on the command line (/?) to display the
supported command line paramaters

twangboy committed:
Add documenting comments

twangboy committed:
Update documentation with new changes to installer

twangboy committed:
Fix typo

twangboy committed:
Fix some grammer issues in the docs

twangboy committed:
Add release notes

twangboy committed:
Fix bad headings

twangboy committed:
Fix `unit.templates.test_jinja` for Windows

Fix problem with files that end with new line on Windows
Fix some problems with regex
Fix some unicode conversion issues

twangboy committed:
Fix `unit.test_pillar` for Windows

Use delete=True in NamedTemporaryFile command otherwise the handle to
the file isn't released and the file can't be opened by the actual test.

gtmanfred committed:
convert jinja context to dictionary

For `show_full_context`, it is useful to have this as a serializable dictionary so that it can be dumped and actually parsed using

`{{show_full_context|yaml(False)}}`

Commit:
Adapted documentation of delete_on_termination parameter

Commit:
[log/sentry] avoid KeyError: 'SENTRY_PROJECT'

if compute_scope is not present on transport registry, we can't
update options dictionnary with empty dsn_config dictionary

fix  #43949

pratik705 committed:
create route to send traffic by natting for_GCP

A route is a rule that specifies how certain packets should be handled by the
virtual network. Routes are associated with virtual machine instances by tag,
and the set of routes for a particular VM is called its routing table.
For each packet leaving a virtual machine, the system searches that machine's
routing table for a single best matching route.

This module will create a route to send traffic destined to the Internet
through your gateway instance.

Commit:
Bugfix: always return a string "list" on unknown job target type.

terminalmage committed:
Move 13 functions from salt.utils to salt.utils.data

These functions are:

- salt.utils.compare_dicts
- salt.utils.compare_lists
- salt.utils.decode_dict
- salt.utils.decode_list
- salt.utils.exactly_n
- salt.utils.exactly_one
- salt.utils.traverse_dict
- salt.utils.filter_by
- salt.utils.traverse_dict_and_list
- salt.utils.subdict_match
- salt.utils.substr_in_list
- salt.utils.is_dictlist
- salt.utils.repack_dictlist

terminalmage committed:
Move 4 functions from salt.utils

- salt.utils.reinit_crypto -> salt.utils.crypt.reinit_crypto
- salt.utils.appendproctitle -> salt.utils.process.appendproctitle
- salt.utils.daemonize -> salt.utils.process.daemonize
- salt.utils.daemonize_if -> salt.utils.process.daemonize_if

benediktwerner committed:
Renamed digitalocean cloud module doc file

msummers42 committed:
addresses issue #43307, disk.format_ to disk.format

This change fixes breakage. It appears the disk.format_ func is
aliased to disk.format in modules/disk.py

vernondcole committed:
Merge remote-tracking branch 'origin/vagrant-module' into vagrant-module

Commit:
Adding requirement for requests 2.4.2 when using Vault module

nasenbaer13 committed:
Merge branch 'develop' into boto_asg

Commit:
Merge pull request #43887 from techhat/azuredep

Add doc for new requirement (azurearm)

Commit:
Merge pull request #43769 from mtorromeo/git-noexec-fix

Copy git ssh-id-wrapper to /tmp only if necessary (Fixes #10582, #19532)

Commit:
Merge branch 'develop' into vmm_grain

Commit:
Merge pull request #43952 from jdsieci/2017.7_add_requisites_to_stateconf

add requisites to stateconf ( backport #43920)

Commit:
Merge pull request #43946 from jerrykan/apt_key_text_check

Document key_text option in pkgrepo state and add some checks

Commit:
Include versionadded tags for new key_text kwarg

Refs #43946 and #40040

terminalmage committed:
Remove docstrings in salt.utils convenience functions

Having the duplicate docstrings makes CodeClimate cry.

Ch3LL committed:
fix 2016.3.7 release notes merge conflict

Ch3LL committed:
fix 2016.3.7 release notes merge conflict

garethgreenaway committed:
Merge branch 'develop' into vagrant-module

garethgreenaway committed:
Adding the ability to pass pillar values in when using state.sls_id

garethgreenaway committed:
Updating docstring with right versionadded information

unthought committed:
Fix GCE provider: #create returns bootstrap result

fixes #43997

dis-xcom committed:
Fix issue with mutable 'virt:disk' object

Images stored in 'virt:disk:<profile>:image' , if exist, have
higher priority than 'image' argument for 'virt' module.

But if different nodes will use different images for the same disk
<profile> during the run, then the image used for the first node
will be re-used for all the rest because it is stored in the
mutable object.

Use copy.deepcopy to avoid such issue.

terminalmage committed:
Fix on_header callback when not redirecting and no Content-Type present

Commit:
Merge pull request #43962 from bobrik/kmod-built-in

Report built-in modiles in kmod.available, fixes #43945

terminalmage committed:
No need to set a specific encoding if one hasn't been provided via the headers

benediktwerner committed:
Added missing tutorial docs to the tutorial index

benediktwerner committed:
Fixed spelling mistake in salt_bootstrap tutorial

garethgreenaway committed:
Merge branch 'develop' into fix-sensehat-docs

Commit:
Merge pull request #43955 from meaksh/2016.11-fix-2291

Enable a new '--with-salt-version' parameter for the "setup.py" script

whiteinge committed:
Also catch cpstats AttributeError for bad CherryPy release ~5.6.0

benediktwerner committed:
Fixec back-ticks in sensehat module

garethgreenaway committed:
This should be mount_opts and I typoed it as opts.  Fixing that.

Commit:
Merge pull request #43993 from benediktwerner/fix-sensehat-docs

Improved sensehat execution module documentation

Ch3LL committed:
Add 2016.11.9 Release Note File

Ch3LL committed:
Add 2017.7.3 Release Note File

Commit:
Merge pull request #44019 from benediktwerner/2016.11

Added missing docs to the tutorial index and fixed  spelling mistake

Commit:
Merge pull request #44018 from benediktwerner/develop

Fixed digitalocean cloud module doc file not being renamed by #43588

Commit:
Merge pull request #44016 from terminalmage/issue44013

Fix on_header callback when not redirecting and no Content-Type present

Commit:
Merge pull request #43998 from unthought/fix-gce-make_master

Fix gce make_master

Commit:
Merge pull request #44023 from Ch3LL/11.9rn

Add 2016.11.9 Release Note File

Commit:
Merge pull request #44024 from Ch3LL/7.3rn

Add 2017.7.3 Release Note File

dayid committed:
Typo correction of lover to lower

Commit:
Merge pull request #44012 from Ch3LL/2017.7.1_follow_up

Security Fixes for 2017.7.2

Commit:
Merge pull request #44011 from Ch3LL/2016.11.7_follow_up

Security Fixes for 2016.11.8

Commit:
Merge pull request #44010 from Ch3LL/2016.3.7_follow_up

Security Fixes for 2016.3.8

Commit:
Merge pull request #44006 from rallytime/follow-up-43946

Include versionadded tags for new key_text kwarg

dayid committed:
Merge branch '2016.11' of https://github.com/saltstack/salt into lover_typo

Commit:
Merge branch '2016.3' into '2016.11'

No conflicts.

Commit:
Merge branch '2016.11' into '2017.7'

No conflicts.

terminalmage committed:
Move 5 functions from salt.utils

These functions are:

- salt.utils.ip_bracket -> salt.utils.zeromq.ip_bracket
- salt.utils.gen_mac -> salt.utils.network.gen_mac
- salt.utils.mac_str_to_bytes -> salt.utils.network.mac_str_to_bytes
- salt.utils.refresh_dns -> salt.utils.network.refresh_dns
- salt.utils.dns_check -> salt.utils.network.dns_check

garethgreenaway committed:
Merge branch 'develop' into vagrant-module

garethgreenaway committed:
Merge pull request #43600 from vernondcole/vagrant-module

Vagrant module

vernondcole committed:
utility support for vagrant-cloud

vernondcole committed:
vagrant states

vernondcole committed:
vagrant cloud driver

vernondcole committed:
documentation for vagrant cloud & state

vernondcole committed:
documentation for vagrant cloud & state

Commit:
Merge pull request #44030 from rallytime/merge-2016.11

[2016.11] Merge forward from 2016.3 to 2016.11

Commit:
Lint: Fixup undefined variable errors

Commit:
Merge pull request #44027 from rallytime/bp-44012

Back-port #44012 to 2017.7.2

garethgreenaway committed:
Merge pull request #44022 from garethgreenaway/mount_state_module_fix

[develop] Fix to mount state module

garethgreenaway committed:
Merge pull request #44026 from rallytime/bp-43950

Back-port #43950 to 2017.7

Commit:
Merge pull request #43982 from terminalmage/salt.utils.data

Move 13 functions from salt.utils to salt.utils.data

Commit:
Merge pull request #44025 from dayid/lover_typo

Typo correction of lover to lower

Commit:
Merge branch 'develop' into bugs/43003_group_syndic_develop

Commit:
Merge pull request #43053 from DSRCorporation/bugs/43003_group_syndic_develop

Group minion events in Syndic.

terminalmage committed:
Fix bad copypasta in warning message

Commit:
Merge pull request #44049 from terminalmage/fix-warning

Fix bad copypasta in warning message

Commit:
Merge pull request #44045 from isbm/isbm-tgttype-fix-2017-port

Bugfix: always return a string "list" on unknown job target type.

Commit:
Merge pull request #44031 from rallytime/merge-2017.7

[2017.7] Merge forward from 2016.11 to 2017.7

twangboy committed:
Fix some docs in the win_dacl state module

Commit:
Merge pull request #43924 from kunal-bajpai/fix_id_cache

Create minion_id dirs if not exists

Commit:
Merge pull request #43897 from ablinkinz/centurylinkcloud_provider

Adding Century Link Cloud Provider

Commit:
Merge pull request #43831 from eradman/vmm_grain

Detect OpenBSD guest running under VMM(4)

Commit:
Merge pull request #43889 from CorvinM/issue43449

Fix issue with using roster_defaults with flat or cloud rosters.

kkoch986 committed:
change tags to list based in config

kkoch986 committed:
Merge branch 'develop' of github.com:kkoch986/salt into develop

mkoderer committed:
Use correct mac prefix for kvm/qemu

Using the private mac range for vms makes it impossible to
distinguish between hypervisors.

Issues #44056

Signed-off-by: Marc Koderer <marc.koderer@sap.com>

The-Loeki committed:
doc fixes for salt-ssh roster(s) opts

Commit:
Add option to eagerly scrub the disk

hackacad committed:
modules elbv2 fix

Ch3LL committed:
Add Known CherryPy Issue to 2017.7.2 Release Notes

Commit:
Merge pull request #43989 from terminalmage/salt.utils.process

Move 4 functions from salt.utils

Commit:
Merge pull request #44066 from Ch3LL/cherry_release

Add Known CherryPy Issue to 2017.7.2 Release Notes

Commit:
Merge branch '2017.7' into jid_queue

Commit:
Merge pull request #43849 from kkoch986/develop

Add support for tagging newly created DigitalOcean droplets.

Commit:
Merge pull request #44051 from twangboy/win_fix_docs_dacl

Fix some documentation formatting issues in the win_dacl state

Commit:
Merge pull request #44002 from rawkode/feature/add-requests-warning

Adding requirement for requests 2.4.2 when using Vault module

Commit:
Merge pull request #44017 from dis-xcom/develop

Fix issue with mutable 'virt:disk' object

garethgreenaway committed:
Merge branch 'develop' into fix_returner_core

garethgreenaway committed:
Various fixes to beacons to handle situations when beacons are stored in pillar.  Updating execution module to report that beacons stored in pillar can not be managed via module.  Updating add function to ensure beacon is available before attempting to add it.  Updaitng unit tests to include new beacon is available check.

hackacad committed:
boto_elbv2 fix

Commit:
Merge pull request #44047 from terminalmage/salt.utils.network

Move 5 functions from salt.utils

Commit:
Merge pull request #44021 from whiteinge/cpstats-attribute-error

Also catch cpstats AttributeError for bad CherryPy release ~5.6.0

terminalmage committed:
Move 14 more functions from salt.utils

These functions are:

salt.utils.test_mode -> salt.utils.args.test_mode
salt.utils.split_input -> salt.utils.args.split_input
alt.utils.pem_finger -> salt.utils.crypt.pem_finger
salt.utils.is_bin_file -> salt.utils.files.is_binary
salt.utils.list_files -> salt.utils.files.list_files
salt.utils.safe_walk -> salt.utils.files.safe_walk
salt.utils.st_mode_to_octal -> salt.utils.files.st_mode_to_octal
salt.utils.normalize_mode -> salt.utils.files.normalize_mode
salt.utils.human_size_to_bytes -> salt.utils.files.human_size_to_bytes
salt.utils.get_hash -> salt.utils.hashutils.get_hash
salt.utils.is_list -> salt.utils.data.is_list
salt.utils.is_iter -> salt.utils.data.is_iter
salt.utils.isorted -> salt.utils.data.sorted_ignorecase
salt.utils.is_true -> salt.utils.data.is_true

hackacad committed:
Merge branch 'develop' into develop

Shinzu committed:
Added version tag

Commit:
fix typo in set_volume_tags documentation

vernondcole committed:
lint and documentation fixes

Ch3LL committed:
Add spm shell tests

skizunov committed:
opkg: Fix usage with pkgrepo.managed

Currently, when using `pkgrepo.managed`, the following error is reported:

```
TypeError: get_repo() got an unexpected keyword argument 'ppa_auth'
```

Fix this by adding `**kwargs` to `get_repo` so that it will absorb unused
kwargs instead of erroring out when given an unknown kwarg. Did the same
thing for all other public APIs in this file.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

Commit:
Merge pull request #44074 from SaltyCharles/patch-9

fix typo in set_volume_tags documentation

terminalmage committed:
Convert "Can be removed" lines into TODO lines

garethgreenaway committed:
Some cleanup of duplicated code.  Fixing event comments.

vernondcole committed:
convert netapi calls to client calls

vernondcole committed:
remove references to salt-api from docs

hackacad committed:
boto_elbv2 fix

hackacad committed:
Merge branch 'develop' of https://github.com/saltstack/salt into develop

hackacad committed:
Merge branch 'develop' of https://github.com/netzmacher/salt into develop

Commit:
Fixing a typo in the troubleshooting document

nasenbaer13 committed:
Merge branch '2017.7' into boto_lc

cetanu committed:
Catch on empty Virtualbox network addr #43427

Commit:
Merge pull request #44063 from netzmacher/develop

modules elbv2 fix

Commit:
Merge branch 'develop' into patch-1

Commit:
Merge pull request #44061 from Shinzu/develop

Add option to eagerly scrub the disk

pratik705 committed:
Fixed "create_attach_volumes" salt-cloud action for GCP

Fixes #44088
    - "create_attach_volumes" salt-cloud works with mandatory and
      default arguments as mentioned in docstring of the function.

Commit:
Merge pull request #44084 from senthilkumar-e/doc_update

Fixing a typo in the troubleshooting document

Commit:
Merge pull request #44076 from Ch3LL/spm_test

Add spm shell tests

Commit:
Confusion between host name and address

Keys in target dict are host names (as expanded by clustershell),
not addresses.

Commit:
Merge branch '2017.7' into 2017.7

techhat committed:
Made sure that unicoded data is sent to sha256()

Commit:
Merge branch 'develop' into develop

Commit:
Merge pull request #44054 from eyj/boto_lc

Backport of missing delete_on_termination

techhat committed:
Use system encoding

techhat committed:
Apparently __salt_system_encoding__ is a thing

techhat committed:
One more encoding

garethgreenaway committed:
Clarifying documention for pillar kwarg for sls and sls_id.

garethgreenaway committed:
Merge branch '2017.7' into develop2

Commit:
Merge pull request #44075 from terminalmage/salt.utils.files

Move 14 more functions from salt.utils

garethgreenaway committed:
Merge branch 'develop' into vagrant-cloud

garethgreenaway committed:
Merge pull request #44092 from techhat/awsunicode

Made sure that unicoded data is sent to sha256()

garethgreenaway committed:
Merge branch 'develop' into patch-1

twangboy committed:
Return multiprocessing queue in LogSetupMock class

twangboy committed:
Skip Master, Minion, and Syndic parser tests

pratik705 committed:
add "auto_delete" option to "attach_disk" and "create_attach_volumes"
for GCP.

fixes #44101

  - This fix will allow user to specify "auto_delete" option with
    attach_disk or "create_attach_volumes" function

terminalmage committed:
Complete the salt.utils refactor

This moves the remaining 30 functions from salt.utils to new locations.

roaldnefs committed:
Fix doc indentation in mattermost_returner

terminalmage committed:
Use one salt.utils.gitfs.GitFS instance per thread

This reduces some of the overhead caused by many concurrent fileclient
requests from minions.

Additionally, initializing remotes has been folded into the GitBase
class' dunder init. Instantiating an instance and initializing its
remotes were initially split so that an object could simply be created
with the master opts so that the configuration was loaded and nothing
else, allowing for certain cases (like clearing the cache files) where
we didn't need to actually initialize the remotes. But this both A)
presents a problem when the instance being used is a singleton, as you
don't want to be re-initializing the remotes all the time, and B)
suppressing initialization can be (and is now being) done via a new
argument to the dunder init.

nasenbaer13 committed:
Merge branch '2017.7' into jid_queue

sahama committed:
show_all remove

show_all=True,  removed from beacon execution module so it should remove here also

Commit:
Merge branch '2016.11' into '2017.7'

Conflicts:
  - salt/utils/aws.py

Commit:
Merge pull request #44106 from roaldnefs/fix-doc-mattermost_returner

Fix mattermost returner documentation

Commit:
Reduce the number of days an issue is stale by 25

msummers42 committed:
fixed test addressing issue #43307, disk.format_ to disk.format

benediktwerner committed:
Fixed code snippet in unit testing doc

Commit:
Merge pull request #44099 from twangboy/win_skip_test_parsers

Skip Master, Minion, and Syndic parser tests

Commit:
Merge pull request #44091 from quinot/bug/clustershell-host-addr-confusion

Confusion between host name and address

Commit:
Merge pull request #44118 from rallytime/merge-2017.7

[2017.7] Merge forward from 2016.11 to 2017.7

Commit:
Merge branch '2017.7.2' into '2017.7'

Conflicts:
  - doc/topics/releases/2017.7.2.rst

Commit:
Merge pull request #44103 from terminalmage/salt.utils

Complete the salt.utils refactor

Commit:
Merge pull request #44098 from twangboy/win_mock_test_parsers

Return multiprocessing queue in LogSetupMock class

Commit:
Merge pull request #44121 from benediktwerner/2017.7

Fixed code snippet in unit testing documentation

Commit:
Merge pull request #44117 from sahama/patch-1

show_all remove

terminalmage committed:
Add top-level docstrings to a bunch of salt.utils modules which are missing them

Commit:
Merge pull request #44015 from gtmanfred/develop

convert jinja context to dictionary

nasenbaer13 committed:
Merge branch '2017.7' into jid_queue

Commit:
Merge pull request #44090 from pratik705/fix-create_attach_volumes_salt-cloud_action-GCP

Fix create_attach_volumes salt-cloud action for gcp

Commit:
Merge pull request #44119 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 25

Commit:
Merge branch '2017.7' into 'develop'

Conflicts:
  - doc/topics/tutorials/index.rst
  - salt/crypt.py
  - salt/modules/cmdmod.py
  - salt/modules/grains.py
  - salt/pillar/git_pillar.py
  - salt/roster/clustershell.py
  - tests/unit/states/test_augeas.py
  - tests/unit/utils/test_utils.py

Commit:
Update any moved utils references from the merge-forward

Commit:
Merge pull request #44079 from skizunov/develop2

opkg: Fix usage with pkgrepo.managed

Commit:
Merge pull request #44029 from msummers42/2017.7

addresses issue #43307, disk.format_ to disk.format

Commit:
Merge pull request #44125 from rallytime/merge-2017.7

[2017.7] Merge forward from 2017.7.2 to 2017.7

Commit:
Merge pull request #44060 from The-Loeki/doc-roster

doc fixes for salt-ssh roster(s) opts

Commit:
Merge pull request #44055 from eyj/jid_queue

Activate jid_queue also for SingleMinions to workaround (Backport)

Commit:
Merge pull request #44127 from terminalmage/salt.utils

Add top-level docstrings to a bunch of salt.utils modules which are missing them

garethgreenaway committed:
Updating error messages to be more clear when beacons are configured in pillar.  Updating the list to ensure the parameters are clear.

garethgreenaway committed:
Updating some formatting.

Commit:
Fix typos in paralell states docs

Commit:
Merge pull request #44129 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

timfreund committed:
Insert missing verb in gitfs walkthrough

nasenbaer13 committed:
Merge branch 'develop' into boto_asg

kstreee committed:
Fixes inconsistent exception handling in rest_tornado.

Commit:
Merge pull request #44135 from timfreund/insert_missing_verb

Insert missing verb in gitfs walkthrough

wido committed:
parted: Cast boundary to string when checking unit type

It could be that just a number (integer) is passed to as either
start or end to the module which is then already a int.

The endswith() method will then fail because that is a String function.

By casting it to a String while checking we make sure that the function
works.

Signed-off-by: Wido den Hollander <wido@widodh.nl>

Commit:
Merge pull request #44133 from cachedout/fix_paralell_docs

Fix typos in parallel states docs

Commit:
Merge branch 'develop' into add-auto_delete-option-while-attaching_disk

b-harper committed:
add documentation for venv_bin in virtualenv_mod state

Commit:
Merge branch 'develop' into gitfs-singleton

Commit:
Merge pull request #44037 from vernondcole/vagrant-cloud

Vagrant cloud and state modules

Commit:
Merge branch 'develop' into 43961_sls_id_pillar

Commit:
Merge pull request #44014 from garethgreenaway/43961_sls_id_pillar

[develop] Additions to state.sls_id

Commit:
Merge pull request #44058 from mkoderer/issues/44056

Use correct mac prefix for kvm/qemu

Commit:
Merge pull request #43917 from twangboy/win_unit_test_pillar

Fix `unit.test_pillar` for Windows

Commit:
Merge pull request #43913 from twangboy/win_fix_test_jinja

Fix `unit.templates.test_jinja` for Windows

cachedout commented:
@almoore I'm not following how the changes to these two files relate to each other. Could you please clarify?

Commit:
Replace salt utils import with correct salt.utils.files path

This fixes a couple of test failures in the develop branch and cleans up
a couple of formatting issues and import ordering.

Commit:
Merge branch '2017.7' into 'develop'

Conflicts:
  - salt/modules/opkg.py
  - salt/utils/templates.py
  - tests/unit/templates/test_jinja.py

Commit:
Update to_unicode util references to new stringutils path

garethgreenaway committed:
Merge pull request #44146 from rallytime/groupadd-cleanup

Replace salt utils import with correct salt.utils.files path

benediktwerner committed:
Skip shadow module unit test if not root or no shadow

Commit:
Merge pull request #44141 from wido/parted-units

parted: Cast boundary to string when checking unit type

Commit:
Merge pull request #44144 from b-harper/doc-venv_bin-state

add documentation for venv_bin in virtualenv_mod state

Commit:
Merge pull request #44123 from terminalmage/gitfs-singleton

Use one salt.utils.gitfs.GitFS instance per thread

Commit:
Merge pull request #44102 from pratik705/add-auto_delete-option-while-attaching_disk

add "auto_delete" option to "attach_disk" and "create_attach_volumes"

Commit:
Merge pull request #44089 from cetanu/patch-1

Catch on empty Virtualbox network addr #43427

Commit:
Merge pull request #44153 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

Commit:
Merge branch 'develop' into boto_asg

Commit:
Reorder and group imports in jina template unit tests

The imports were scattered around quite a bit, which made it hard
to see where multiple uses of salt/utils/* imports were being pulled
in.

This PR just adjusts the ordering so it's easier to see what is included
already for the future. (This is particularly useful for catching misuses
of salt.utils.__init__.py functions.)

Commit:
Merge pull request #43454 from kunal-bajpai/fix_returner_core

Fix save_load which is never called for returner jobs

benediktwerner committed:
Fixed missing import

vernondcole committed:
add more unit tests

vernondcole committed:
remove unused parameter

Commit:
Fixing default redis.host in documentation

Commit:
Merge pull request #44159 from benediktwerner/fix-shadow-test

Fix shadow module unit test if not root or the module is not present

Commit:
Merge pull request #43649 from twangboy/win_installer_multimaster

Add support for multimaster setup

Commit:
Merge pull request #44177 from senthilkumar-e/redis_doc_update

Fixing default redis.host in documentation

Commit:
Fixup warn_until messages

This change also updates the `salt.utils.warn_until` path to the new `salt.utils.versions.warn_util` path.

Commit:
Whitespace fix

vernondcole committed:
lint and comment fixes

Commit:
Merge branch 'develop' into ssh_known_hosts_improvements

Commit:
Merge pull request #44176 from vernondcole/simplify-saltify

Simplify saltify cloud driver

Commit:
Merge branch 'develop' into jinja-test-imports

Commit:
Merge branch 'develop' into fix-inconsistent-exception-handling

Commit:
Merge branch 'develop' into various_beacon_fixes

cachedout commented:
Bump @almoore

Commit:
Merge pull request #43966 from pratik705/create-route-to-send-traffic-by-natting_for_GCP

create route to send traffic by natting for_GCP

kstreee committed:
Merge branch 'develop' into fix-inconsistent-exception-handling

Commit:
Merge pull request #43476 from Shados/ssh_known_hosts_improvements

Improvements to SSH known_hosts handling

Commit:
Merge pull request #43858 from eyj/boto_asg

Boto_ASG fixes for scaling policy rate limiting and tag conversion

Commit:
Merge pull request #44162 from rallytime/jinja-test-imports

Reorder and group imports in jina template unit tests

Commit:
Merge pull request #43821 from bcelenk/opsgenie-modules

Add OpsGenie's execution and state modules

Commit:
Merge pull request #44085 from kstreee/fix-inconsistent-exception-handling

Fixes inconsistent exception handling in rest_tornado.

Commit:
Merge pull request #43838 from bodhi-space/infra5669_2

INFRA-5669 - add retry loops for API rate limiting in ELB health_check funcs

Commit:
Reduce the number of days an issue is stale by 25

Commit:
Merge pull request #44071 from garethgreenaway/various_beacon_fixes

[develop] Various fixes to beacons

garethgreenaway committed:
Merge pull request #44204 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 25

almoore commented:
The two commits are not related. The one with the jsonb filtering was pushed up on accident. I was trying to mark it as not related but was having an issue with that. I can take that out now.

almoore committed:
Merge branch 'develop' of github.com:almoore/salt into develop

almoore committed:
Revert changes to pgjsonb from 553aaad105434d25c982a90b16b8387163d89ffa
